vumi/middleware/session_length.py

```diff
@@ -0,0 +1,87 @@
+# -*- test-case-name: vumi.middleware.tests.test_session_length -*-
+
+from twisted.internet.defer import inlineCallbacks, returnValue
+from twisted.internet import reactor
+
+from vumi.message import TransportUserMessage
+from vumi.middleware.base import BaseMiddleware
+from vumi.persist.txredis_manager import TxRedisManager
+
+
+class SessionLengthMiddleware(BaseMiddleware):
+    """ Middleware for storing the session length in the message.
+
+    Stores the start timestamp of a session in the message payload in
+    message['helper_metadata'][field_name]['session_start'] for all messages
+    in the session, as well as the end timestamp if the session in
+    message['helper_metadata'][field_name]['session_end'] if the message
+    marks the end of the session.
+
+    Configuration option:
+
+    :param dict redis:
+        Redis configuration parameters.
+    :param int timeout:
+        Redis key timeout (in seconds). Defaults to 120.
```

Comment:
Session can be a bit longer than this so I'd prefer a timeout of, say, ``600`` seconds?